### PR TITLE
feat: introduce a captions menu button

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by gkatsev** [comment]
The tests will fail until https://github.com/muxinc/media-chrome/pull/442 is merged and available

File: packages/mux-player/src/styles.css [hunk 1]
```diff
@@ -104,6 +112,10 @@ a {
   --media-seek-forward-button: var(--seek-forward-button, var(--bottom-seek-forward-button));
   --media-mute-button: var(--mute-button, var(--bottom-mute-button));
   --media-captions-button: var(--captions-button, var(--bottom-captions-button));
+  --media-captions-menu-button: var(
+    --captions-menu-button,
+    var(--media-captions-button, var(--bottom-captions-menu-button))
+  );
```
Comment (luwes) [inline packages/mux-player/src/styles.css:118]: 👍

File: packages/mux-player/src/media-theme-mux.html [hunk 2]
```diff
@@ -69,8 +69,21 @@
...
+    media-controller:not(:is([media-captions-list], [media-subtitles-list])) media-captions-menu-button {
+      --captions-menu-button: none;
+    }
+
+    media-captions-menu-button {
+      --media-listbox-background: var(--secondary-color);
+      --media-listbox-selected-background: rgba(255, 255, 255, 0.28);
+      --media-listbox-hover-background: none;
+      --media-listbox-hover-outline: white solid 1px;
+      --media-text-color: white;
```
Comment (luwes) [inline packages/mux-player/src/media-theme-mux.html:86]: 👍

File: packages/mux-player/src/media-theme-mux.html [hunk 3]
```diff
@@ -383,10 +396,10 @@
       </media-pip-button>
     </template>
 
-    <template partial="CaptionsButton">
-      <media-captions-button
+    <template partial="CaptionsMenuButton">
+      <media-captions-menu-button
         default-showing="{{defaultShowingCaptions}}"
-        part="bottom captions button"
+        part="bottom captions menu-button"
```
[the remark this answers is not in the document]
Comment (gkatsev) [reply packages/mux-player/src/media-theme-mux.html:402]: this gets a bit complicated, especially with the current implementation of the captions-menu-button, where you're styling a parent container which has `display:contents` and aren't styling the button itself.

[the remark this answers is not in the document]
Comment (luwes) [reply packages/mux-player/src/media-theme-mux.html:402]: hmm okay this is less ideal, should we slot the button then and add the button part in the light DOM for the player theme. 

I'm fine with signing off and following up on this until someone hits this problem.
just seemed not inline with my expectations

[the remark this answers is not in the document]
Comment (gkatsev) [reply packages/mux-player/src/media-theme-mux.html:402]: yeah, this is also part of the issue I brought up on media-chrome and how we want to handle developing nested shadow doms and what not.

**Pull request merged**